build-support/githooks/prepare-commit-msg

```diff
@@ -43,15 +43,15 @@ HAS_WHEELS_SKIP=$?
 if [[ "${HAS_RUST_SKIP}" -eq 1 ]] && [ "${NUM_RUST_FILES}" -eq 0 ]; then
   cat << EOF >> "${COMMIT_MSG_FILEPATH}"
 
-# Rust tests and lints will be skipped. Delete if not intended.
+<!-- Rust tests and lints will be skipped. Delete if not intended. -->
```

Comment:
The loud was sort of the point - you're intended to edit that out. As is the warning is effectively silenced IIUC if someon'e Git workflow does not have them read their local commit final message but does have them mucking with the Github commit message / PR description UI thingy.

Reply:
Yeah, that's where this bit me. I was using a GUI for git operations and the pre-commit hook's commit message concatenation doesn't appear in the GUI. Then in github it gets screamy.

I'm not the only one though... https://github.com/pantsbuild/pants/pulls?q=is%3Apr+%22Delete+if+not+intended%22+is%3Aclosed

I suppose a follow-up question would be: Can we just live without it? Are we worried the logic that puts this in the commit isn't robust enough?

Reply:
Yeah, I've been guilty, but I also view each instance of that as a personal failure (in the light sense). I should have slowed down and followed the process to the end. I think that puts me out of contention for opining on this further though so you should loop others in.

Reply:
I've mentioned to Joshua that as the one who introduced this code, I wish we deleted it and instead had GitHub Actions automatically determine whether to skip things based on inspecting the changed files. Iirc, the only reason we used this label approach is Travis limitations. With GHA, we should be able to have a precursor step that determines which projects are changed, then the other jobs are conditional on that.

In the meantime, I'm generally +1 to this change because it's annoying indeed to have the `#` blow up GH.

Reply:
... That should be trivial actually. I can take a stab at it!

Reply:
> I've mentioned to Joshua that as the one who introduced this code, I wish we deleted it and instead had GitHub Actions automatically determine whether to skip things based on inspecting the changed files. Iirc, the only reason we used this label approach is Travis limitations. With GHA, we should be able to have a precursor step that determines which projects are changed, then the other jobs are conditional on that.
> 
> In the meantime, I'm generally +1 to this change because it's annoying indeed to have the `#` blow up GH.

@Eric-Arellano there is a way to determine which files are changed and decide if we want to skip job steps based on that.
We use that technique internally (in Toolchain) so you can take a look there and maybe apply some of it here.